### ISSUE TITLE
Restructuring of the QUEENS repository

### ISSUE THREAD

**Issue by gilrrei** [queens-py/queens#13]
> This issue was originally created by @gilrrei on [gitlab.lrz.de](https://gitlab.lrz.de/queens_community/queens/-/issues/270) on 2021-12-03.

## Motivation and Context

In order to move towards a more open source worthy QUEENS projects we should do some restructuring of the files. For a pip package (or generally any python application) I propose the following structure ([based on this](https://realpython.com/python-application-layouts/)):
```
queens/
│
├── queens/
│   ├── __init__.py
│   ├── queens.py (replaces main.py)
│   └── other stuff
│
├── tests/
│   ├── benchmarks/
│   ├── unit_tests/
│   ├── integration_tests/
│   └── README.md
│
├── .gitignore
├── LICENSE
├── README.md
├── requirements.txt
├── other stuff-
└── setup.py
```
After closing [266](https://gitlab.lrz.de/queens_community/queens/-/issues/266), the setup and requirements are done, and we can focus on restructuring.

## Possible Solution

- [x] Move `pqueens` to `queens` (makes import nicer `import queens.iterators.lhs_iterator`) (see [612](https://gitlab.lrz.de/queens_community/queens/-/merge_requests/612))
- [x] Bring the `tests` up a level (This way the test are not loaded when importing queens. Strictly speaking they are not part of the source folder anyway) (see [612](https://gitlab.lrz.de/queens_community/queens/-/merge_requests/612))
- [x] Rename `setup_remote.py` to `setup_singularity.py` (or similar) and move it to `queens/utils/` in order to avoid confusion. (or a separate singularity folder would work as well) (see [401](https://gitlab.lrz.de/queens_community/queens/-/merge_requests/401))
- [x] Move `singularity_recipe` to `queens/utils` (or a separate singularity folder would work as well)
- [x] Move the `sobol_sequence.py` to `queens/utils` or create a folder for modifications of other peoples code like for example the modified gradient file from numpy
- [x] Restructure the `queens/doc/source` folder, here are some deprecated files and a jupyter notebook (cool idea but the last update was in March 2018). Also check for different solutions for building the docs. Move to [https://readthedocs.org/](https://readthedocs.org/)?
- [x] fix relative imports e.g. `from .iterator import Iterator`
- [x] Migrate `utils/plot_output` capabilities to `visualitzation module` [263](https://gitlab.lrz.de/queens_community/queens/-/issues/263)
- [x] Remove the module type from the names e.g. `monte_carlo_iterator.py` --> `monte_carlo.py`, `post_post_baci.py`--> `baci.py`. Or if not, at least use a standard.
## Related Issues

* Closes [263](https://gitlab.lrz.de/queens_community/queens/-/issues/263) 
* Is blocked by
* Follows [266](https://gitlab.lrz.de/queens_community/queens/-/issues/266)
* Precedes
* Related to
* Part of
* Composed of

## Additional information
If anyone has other files to move just comment them below.

## Interested Parties

@jnitzler @jbi35 @bwirthl @sbrandstaeter @hwillmann

**Comment by gilrrei** [comment]
Closed with #136
